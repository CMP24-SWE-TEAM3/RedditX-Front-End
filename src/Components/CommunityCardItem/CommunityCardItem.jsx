import {useState} from "react";
import logo from "Assets/logo.png";
import {Container} from "./CommunityCardItem.styled";
/**
 * Component that  shows the names of communities up-to-date.
 * @returns {Component.React}
 */
const CommunityCardItem = ({community}) => {
    const [btnContent, setstate] = useState("Join");

    function clickHandler(e) {
        e.preventDefault();
        if (btnContent === "Join") {
            setstate("Joined");
        } else {
            setstate("Join");
        }
    }

    function mouseEnterHandler() {
        if (btnContent === "Joined") {
            setstate("Leave");
        }
    }

    function MouseLeaveHandler() {
        if (btnContent === "Leave") {
            setstate("Joined");
        }
    }

    return (
        <Container>
            <a href="https://www.reddit.com/r/Steam/">
                <div className="item">
                    <span className={'num'}>1</span>
                    <span className='caret'>&#94;</span>
                    <img src={logo} alt=""/>
                    <div className="info">
                        <div className="info2">
                            <h6>r/{community}</h6>
                        </div>
                    </div>
                    <div className="button">
                        <button
                            onClick={clickHandler}
                            onMouseEnter={mouseEnterHandler}
                            onMouseLeave={MouseLeaveHandler}>
                            {btnContent}
                        </button>
                    </div>
                </div>
            </a>
        </Container>
    );
};

export default CommunityCardItem;