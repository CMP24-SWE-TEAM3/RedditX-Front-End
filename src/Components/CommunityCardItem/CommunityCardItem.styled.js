import styled from "styled-components";

export const Container = styled.div`
  
  & a {
    margin-bottom: 0;
      //border-bottom: thin solid ${({theme}) => theme.lineColor.PostBorder};
    border-radius: 4px 4px 0 0;
    //padding-top: 16px;
    display: flex;
    flex-direction: column;
    margin-top: -1px;
    padding: 16px;
    color: inherit;
    text-decoration: inherit;
  }

  & .item span {
    color: ${({theme}) => theme.color.primary};
    font-weight: 500;
    line-height: 18px;
  }

  .item .caret {
    color: rgb(70 209 96);
    font-size: 20px;
    font-weight: 400;
    height: 20px;
    line-height: 20px;
    vertical-align: middle;
    width: 20px;
    margin-left: 8px;
  }

  & .item {
    align-items: center;

    display: flex;

    justify-content: space-between;
  }

  & .item img {
    border: none;
    background-color: #205493;
    font-size: 36px;
    height: 36px;
    max-height: 36px;
    max-width: 36px;
    overflow: hidden;
    width: 36px;
    line-height: normal;
    //-ms-flex-negative: 0;
    //margin: 0 8px;
    flex-shrink: 0;
    border-radius: 100%;
    vertical-align: middle;
  }

  & .info {
    flex-grow: 1;
    padding: 0 8px;
    overflow: hidden;
    overflow-wrap: break-word;
  }

  & .info2 {
    display: flex;
    align-items: baseline;
    flex-direction: column;
  }

  & .info2 h6 {
    font-size: 12px;
    //font-weight: 500;
    line-height: 16px;
    color: ${({theme}) => theme.color.primary};
    font-weight: 700;
    max-width: 100%;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  & .info2 h6:hover {
    text-decoration: none;
  }

  & .info2 p {
    font-size: 12px;
    font-weight: 400;
    line-height: 16px;
    color: ${({theme}) => theme.color.muted};
  }

  & .button {
    flex-shrink: 0;
    //width: 88px;
    border: none;
  }

  & button {
    position: relative;
    background-color: ${({theme}) => theme.color.secondary};
    border: 1px solid transparent;
    color: ${({theme}) => theme.background.primary};
    fill: ${({theme}) => theme.color.secondary};
    font-family: Noto Sans, Arial, sans-serif;
    font-size: 12px;
    font-weight: 700;
    letter-spacing: unset;
    line-height: 17px;
    text-transform: unset;
    height: 24px;
    width: 56px;
    padding: 4px 16px;
    align-items: center;
    border-radius: 9999px;
    box-sizing: border-box;
    display: flex;
    justify-content: center;
    text-align: center;
    cursor: pointer;
  }

  & button:hover {
    background-color: ${({theme}) => theme.background.hover_background_button_blue};
  }

  & .num {
    color: ${({theme}) => theme.color.primary};
    text-align: right;
    min-width: 20px;
    font-size: 14px;
    font-weight: 500;
    line-height: 18px;
    
  }
`