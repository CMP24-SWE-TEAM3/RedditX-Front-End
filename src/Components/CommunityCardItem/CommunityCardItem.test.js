import {shallow} from 'enzyme';

// import Components
import CommunityCardItem from "./CommunityCardItem";

describe("Community Card Item Component", () => {
    it("this is a test for Community Card Item Component", () => {
        expect(shallow(<CommunityCardItem/>)).toMatchSnapshot();

    });
});