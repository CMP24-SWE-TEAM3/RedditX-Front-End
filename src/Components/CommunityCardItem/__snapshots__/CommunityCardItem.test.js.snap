// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Community Card Item Component this is a test for Community Card Item Component 1`] = `ShallowWrapper {}`;
